common/src/JetIds.cxx

```diff
@@ -92,20 +93,40 @@ bool JetPFID::looseID(const Jet & jet) const{
   return false;
 }
 
+
+//according to https://twiki.cern.ch/twiki/bin/viewauth/CMS/JetID13TeVRun2017
 bool JetPFID::tightID(const Jet & jet) const{
-  if(!looseID(jet)) return false;
-  if(fabs(jet.eta())<=2.7 
-     && jet.neutralEmEnergyFraction()<0.90
-     && jet.neutralHadronEnergyFraction()<0.90){ //2016 jetID: "tight" differ from "loose" only at |eta|<2.7
+    if(fabs(jet.eta())<=2.7
+     && jet.numberOfDaughters()>1 
+     && jet.neutralHadronEnergyFraction()<0.90
+     && jet.neutralEmEnergyFraction()<0.90){
+    
+    if(fabs(jet.eta())>=2.4)
+      return true;
+      
+    if(jet.chargedHadronEnergyFraction()>0
+       && jet.chargedMultiplicity()>0)
+      return true;   
+  }
+  else if(fabs(jet.eta())>2.7 && fabs(jet.eta())<=3
+	  &&jet.neutralEmEnergyFraction()<0.90
```

Comment:
This should be 0.99 according to the twiki page

Reply:
you are right, I corrected this